### DISCUSSION TITLE
security nextcloud service

### DISCUSSION THREAD

**Discussion by MorganReid** [keycloak/keycloak#19840] [Q&A] [answered]
I configured according to [stack overflow - SSO with SAML, Keycloak and Nextcloud](https://stackoverflow.com/questions/48400812/sso-with-saml-keycloak-and-nextcloud), my realm name is `myrealm` and my keycloak host is 
`http://10.112.0.141:51122/ `.but an error was reported during the final test：

![Screenshot from 2023-04-20 19-14-03.png](https://comment.janikvonrotz.ch/api/v1/picture/github_98b53c487c407a2b7466456634c56ac2b0ce7ec8/ch0i21v357592gmabkfg)
The keycloak I deployed based on the latest version (21.1.0) of the mirror cannot be found.So during the configuration process, I did not operate this step, will it affect it?
``Update the Client SAML Endpoint field with: https://login.example.com/auth/realms/example.com.``

besides,If I replace `/auth `with `/admin` in the path in the configuration below:
`Identifier of the IdP: https://login.example.com/auth/realms/example.com
URL Target of the IdP where the SP will send the Authentication Request Message: https://login.example.com/auth/realms/example.com/protocol/saml
URL Location of IdP where the SP will send the SLO Request: https://login.example.com/auth/realms/example.com/protocol/saml`.
I will get another error:

![Screenshot from 2023-04-20 19-21-19.png](https://comment.janikvonrotz.ch/api/v1/picture/github_98b53c487c407a2b7466456634c56ac2b0ce7ec8/ch0i26v357592gmabkg0)
Then I look at the log output of keycloak:
 
![Screenshot from 2023-04-20 19-22-55.png](https://comment.janikvonrotz.ch/api/v1/picture/github_98b53c487c407a2b7466456634c56ac2b0ce7ec8/ch0i2a7357592gmabkgg)
I just came into contact with keycloak, can you help me find out why, thank you very much.

**Comment by darius-m** [comment, marked as answer]
You need to read a bit more of the Keycloak administration documentation to understand how the authentication protocols work, and what the Keycloak endpoints do,

The issue in your case is most likely the fact that in the stackoverflow link that you've linked, they used the old default `/auth` base path, while you are using `/`. As such, the `/auth/realms/myrealm/protocol/saml` is not a valid path, it should be `/realms/myrealm/protocol/saml`. The `/admin` path is used for administration purposes only (with the `/auth` base it would have been `/auth/admin`), so clients should almost never connect to it (unless you explicitly configure some admin privileges to accounts).
